### DISCUSSION TITLE
C++ Lambda 表达式隐式转换引发的 Bug | SF-Zhou's Blog

### DISCUSSION THREAD

**Discussion by SF-Zhou** [SF-Zhou/sf-zhou.github.io#162] [Announcements]
https://sf-zhou.github.io/programming/lambda_implicit_conversion_bug.html

**Comment by SF-Zhou** [comment]
这里就得羡慕一下 Rust 了，它的编译期检查和生命周期管理会比 C++ 更严谨些。

**Comment by seed93** [comment]
踩过这个坑……当时没想为什么……换成不是auto就可以

**Comment by seed93** [comment]
好像不一样。之前你写的threadpool，里面有一个
`std::function<return_type()> task =
            std::bind(std::forward<F>(f), std::forward<Args>(args)...);`
这里原来用auto，在windows下会编译不过

**Comment by SF-Zhou** [comment]
> @seed93
> 好像不一样。之前你写的threadpool，里面有一个
> `std::function<return_type()> task =
>             std::bind(std::forward<F>(f), std::forward<Args>(args)...);`
> 这里原来用auto，在windows下会编译不过

😅
